doc/performance_counters.rst

```diff
@@ -54,16 +56,22 @@ In particular, to enable/disable ``mcycle(h)``, bit 0 must be written. For ``min
 The lower 32 bits of all counters can be accessed through the base register, whereas the upper 32 bits are accessed through the ``h``-register.
 Reads to all these registers are non-destructive.
 
-Counter Config
---------------
+Parametrization at synthesis time
+---------------------------------
+
+The ``mcycle(h)`` and ``minstret(h)`` counters are unconditionally enabled and 64 bit wide.
 
-The ``mcycle(h)`` and ``minstret(h)`` counters are 64 bit wide.
-The bit width of the event counters ``mhpmcounter3(h)`` - ``mhpmcounter31(h)`` can be controlled via the ``WidthMHPMCounters`` parameter.
+The event counters ``mhpmcounter3(h)`` - ``mhpmcounter31(h)`` are parametrizable.
+Their width can be parametrized between 32 and 64 bit through the ``WidthMHPMCounters`` parameter, which defaults to 40 bit wide counters.
 
 The effective number of available event counters ``mhpmcounterX(h)`` can be controlled via the ``NumMHPMCounters`` parameter.
-By default, only the first 8 counters ``mhpmcounter3(h)`` - ``mhpmcounter10(h)`` are available.
-The association of events with these counters is hardwired as listed in the following table.
-The remaining counters are disabled and tied to 0.
+By default, no counters are enabled.
```

Comment:
I would clearly differentiate between available counters and enabled/disabled counters here. Enabled means the counter is counting whereas available means the counter can be enabled/disabled. In the spec, an enabled counter means a counter that is effectively counting up. Enabling/disabling counters it is done in software by writes to the `mcountinhibit` and `mevent3` - `mevent31` CSRs.

Reply:
good point, fixed.